components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/LoadState.kt

```diff
@@ -7,42 +7,87 @@ import androidx.compose.runtime.remember
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.setValue
 
+/**
+ * Represents the load state of [T].
+ */
 sealed class LoadState<T> {
     class Loading<T> : LoadState<T>()
     data class Success<T>(val value: T) : LoadState<T>()
     data class Error<T>(val exception: Exception) : LoadState<T>()
 }
 
+
+/**
+ * Load a block of code with the operation progress and result of [T] wrapped around a [LoadState].
```

Comment:
Maybe "Load an item with type [T] asynchronously, and notify the caller about the load state."